src/circuit/layouter.rs

```diff
@@ -232,9 +232,14 @@ impl<F: Field> RegionLayouter<F> for RegionShape {
         column: Column<Advice>,
         offset: usize,
         constant: Assigned<F>,
-    ) -> Result<Cell, Error> {
+    ) -> Result<AssignedCell<F, F>, Error> {
         // The rest is identical to witnessing an advice cell.
-        self.assign_advice(annotation, column, offset, &mut || Ok(constant))
+        let cell = self.assign_advice(annotation, column, offset, &mut || Ok(constant))?;
+        Ok(AssignedCell {
+            cell,
+            value: None,
```

Comment:
Technically we always have the constant value here, so we should just return it.